common4j/src/main/com/microsoft/identity/common/java/net/HttpResponse.java

```diff
@@ -101,7 +101,7 @@ public String getHeaderValue(@NonNull final String key, final int index){
         }
 
         final List<String> list = mResponseHeaders.get(key);
-        if (list == null){
+        if (list == null || list.size() <= index){
```

Comment:
We should probably also add a check for index being >= 0 (otherwise negative indices might make this crash)